system/Database/BaseResult.php

```diff
@@ -260,7 +260,7 @@ public function getResultObject(): array
      * @param         string                           $type The type of result object. 'array', 'object' or class name.
      * @phpstan-param class-string<T>|'array'|'object' $type
      *
-     * @return         array|object|stdClass|null
+     * @return         array|float|int|object|stdClass|string|null
      * @phpstan-return ($type is 'object' ? stdClass|null : ($type is 'array' ? array|null : T|null))
```

Comment:
Okay.

```suggestion
     * @phpstan-return ($n is string ? float|int|string|null : ($type is 'object' ? stdClass|null : ($type is 'array' ? array|null : T|null)))
```
See https://phpstan.org/blog/phpstan-1-6-0-with-conditional-return-types